rclcpp/include/rclcpp/client.hpp

```diff
@@ -215,6 +216,38 @@ class ClientBase
   bool
   exchange_in_use_by_wait_set_state(bool in_use_state);
 
+  /// Get the actual request publsher QoS settings, after the defaults have been determined.
+  /**
+   * The actual configuration applied when using RMW_QOS_POLICY_*_SYSTEM_DEFAULT
+   * can only be resolved after the creation of the client, and it
+   * depends on the underlying rmw implementation.
+   * If the underlying setting in use can't be represented in ROS terms,
+   * it will be set to RMW_QOS_POLICY_*_UNKNOWN.
+   * May throw runtime_error when an unexpected error occurs.
+   *
+   * \return The actual request publsher qos settings.
+   * \throws std::runtime_error if failed to get qos settings
```

Comment:
shouldn't a `rclcpp::RCLError` be thrown?

Reply:
same applies to the other methods here and in `Server`

Reply:
ah, I see the rcl added functions don't have a return code, so `std::runtime_error` is probably fine